File: src/main/java/com/sy/coladay/security/SecurityConfig.java
package com.sy.coladay.security;

import javax.sql.DataSource;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.authentication.AuthenticationProvider;
import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.config.http.SessionCreationPolicy;
import org.springframework.security.crypto.password.NoOpPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.web.servlet.config.annotation.EnableWebMvc;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

/**
 * Simple security configuration to secure the application. The mechanism used in here like
 * http basic, no password encoder are used to keep the demo simple. They are are not intended for
 * usage in real life scenario.
 *
 * @author selim
 */
@Configuration
@EnableWebSecurity
@EnableWebMvc
@SuppressWarnings("unused")
public class SecurityConfig extends WebSecurityConfigurerAdapter implements WebMvcConfigurer {

  @Autowired
  private UserDetailsServiceImpl userDetailsService;

  @Autowired
  private DataSource dataSource;

  @Override
  protected void configure(HttpSecurity http) throws Exception {

    http
        .sessionManagement()
        .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
        .and()
        // In real life scenario a stronger authentication mechanism must be use to protect the
        // user credentials.
        .httpBasic()
        .and()
        .authorizeRequests()
        .antMatchers("/").permitAll()
        .anyRequest().authenticated()
        .and()
        .csrf()
        .disable();
  }

  @Override
  protected void configure(AuthenticationManagerBuilder auth) {
    auth
        .authenticationProvider(authenticationProvider());
  }

  @Bean
  public AuthenticationProvider authenticationProvider() {
    final DaoAuthenticationProvider authProvider = new DaoAuthenticationProvider();
    authProvider.setUserDetailsService(userDetailsService);
    authProvider.setPasswordEncoder(encoder());
    return authProvider;
  }

  @Bean
  public PasswordEncoder encoder() {
    // For the sake of simplicity we use a simple password encoder. In real life scenario, the
    // password must be at the very least hashed.
    return NoOpPasswordEncoder.getInstance();
  }
}

File: src/main/kotlin/com/sy/coladay/security/SecurityConfig.kt
package com.sy.coladay.security

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.authentication.dao.DaoAuthenticationProvider
import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.config.http.SessionCreationPolicy
import org.springframework.security.crypto.password.NoOpPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder
import javax.sql.DataSource

/**
 * Simple security configuration to secure the application. The mechanism used in here like http
 * basic, no password encoder are used to keep the demo simple. They are are not intended for usage
 * in real life scenario.
 *
 * @author selim
 */
@Configuration
@EnableWebSecurity
open class SecurityConfig : WebSecurityConfigurerAdapter() {
    @Autowired
    private val userDetailsService: UserDetailsServiceImpl? = null

    @Autowired
    private val dataSource: DataSource? = null

    @Throws(Exception::class)
    override fun configure(http: HttpSecurity) {
        http
            .sessionManagement()
            .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
            .and() // In real life scenario a stronger authentication mechanism must be use to protect the
            // user credentials.
            .httpBasic()
            .and()
            .authorizeRequests()
            .antMatchers("/actuator/**").permitAll()
            .anyRequest().authenticated()
            .and()
            .csrf()
            .disable()
    }

    override fun configure(auth: AuthenticationManagerBuilder) {
        val authProvider = DaoAuthenticationProvider()
        authProvider.setUserDetailsService(userDetailsService)
        authProvider.setPasswordEncoder(encoder())
        auth
            .authenticationProvider(authProvider)
    }

    @Bean
    open fun encoder(): PasswordEncoder {
        // For the sake of simplicity we use a simple password encoder. In real life scenario, the
        // password must be at the very least hashed.
        return NoOpPasswordEncoder.getInstance()
    }
}